### ISSUE TITLE
Anistropy messages block execution from script

### ISSUE THREAD

**Issue by mdoube** [bonej-org/BoneJ2#207]
**Describe the bug**
When running from macro, Anisotropy shows messages in the GUI which interrupts execution until the user clicks OK.

**To Reproduce**
Run the macro
```
nDirsMax = 16384;
nLinesMax = 512;
row = 0;
setBatchMode(true);
for (nDirs = 16; nDirs <= nDirsMax; nDirs *= 2){
    for (nLines = 1; nLines <= nLinesMax; nLines *= 2){
		startTime = getTime();
		run("Anisotropy", "inputimage=net.imagej.ImgPlus@73956688 directions="+nDirs+" lines="+nLines+" samplingincrement=1.73 recommendedmin=true printradii=true printeigens=true displaymilvectors=false");
    	endTime = getTime();
    	duration = endTime - startTime;
    	setResult("nDirs", row, nDirs);
    	setResult("nLines", row, nLines);
    	setResult("Duration", row, duration);
    	updateResults();
    	row++;
    }
}
```

**Expected behavior**
The script completes without user interaction. Errors and other messages could be logged to the console of Log window.

**Screenshots**
![image](https://user-images.githubusercontent.com/87103/75943211-59706200-5ecf-11ea-8298-0a44d6d61010.png)

**Additional context**

**Comment by rimadoma** [comment]
Seems to be a [known issue](https://github.com/scijava/scijava-common/issues/317). For now we can 1.Get rid of messages that are not absolutely necessary 2. Add instructions for how to hack the macros.

Option 2 for Anisotropy would look like 
```
run("Anisotropy", "inputimage=net.imagej.ImgPlus@73956688 directions="+nDirs+" lines="+nLines+" samplingincrement=1.73 recommendedmin=true printradii=true printeigens=true displaymilvectors=false, instruction=\"\"");`
```

If you want to go with option 2, we'll need to mark all message parameters with `persist=false`.

**Comment by mdoube** [comment]
option 3: can we just put the message in the dialog box, so that only interactive GUI users see it?

**Comment by rimadoma** [comment]
This is how you add messages to a dialog box. By having
```
@Parameter(visibility = ItemVisibility.MESSAGE)
private String instruction = "Press 'OK' to run the plugin";
```
Like I mentioned above, there just so happens to be bug that makes them pop up when a macro's run.

**Comment by mdoube** [comment]
Option 2. seems to work OK at the moment.

There is one other dialog that pops up:

![image](https://user-images.githubusercontent.com/87103/76175445-98f8c000-61e7-11ea-96b0-1252d7751083.png)

Could we make it so that if code is being called from a script, warnings and messages get output to the console or log window only and not to a GUI OK button window? This particular one doesn't block execution, but you do get a big pile of them accumulating on the screen.

**Comment by rimadoma** [comment]
@alessandrofelder Have you encountered these issues when running Python scripts?

**Comment by alessandrofelder** [comment]
Yea, I think so. I can't remember exactly, but I will try to replicate tomorrow and report here. It was ITA-related, I think.

**Comment by mdoube** [comment]
Fixed by #210 , closing.
